server/caas/api.js

```diff
@@ -0,0 +1,324 @@
+      if (error) {
+        res.status(500).send();
+      } else {
+        if (results.length) {
+          var accountModel = null;
+          var credentialModel = null;
+          newRequest({
+            url: url + '/account',
+            method: 'POST',
+            body: account
+          }, function(response, body) {
+
+            accountModel = body;
+            credential.accountId = body.id;
+
+            newRequest({
+              url: url + '/passwords',
+              method: 'POST',
+              body: credential
+            }, function(response, body) {
+              credentialModel = body;
+              removeUserFromTable(user.email, function(results, fields) {
+                getTokenForLogin(user.email, user.pw, function(response, body) {
+                  res.cookie('token', body.jwt, {secure: true});
+                  res.redirect('/');
+                });
+              }, res);
+            }, res);
+          }, res);
+        } else {
+          res.status(500).send({error: 'Invalid Token'});
+        }
+      }
+    });
+  });
+
+  app.use('/reset-password', function(req, res, next) {
+    var user = req.body;
+    var userEmail = user.email;
+    var name = user.name;
+    var url = config.apiServer + config.apiEndpoint + '/credentials?kind=password&publicValue=' + userEmail + '&limit=-1&sort=name';
+
+    newRequest({
+      url: url,
+      method: 'GET',
+    }, function(response, body) {
+
+      if (body.data.length) {
+
+        var credId = body.data[0].id;
+        var url = config.apiServer + config.apiEndpoint + '/accounts/' + body.data[0].accountId;
+
+        newRequest({
+          url: url,
+          method: 'GET'
+        }, function(response, body) {
+
+          if (body.type === 'account') {
+            generateAndInsertToken(body.id, name, userEmail, "reset", function(error, results, fields, token) {
+              if (error) {
+                res.status(500).send({error: error});
+              } else {
+
+                sendPasswordReset(req.body.email, response.body.name, req.headers.origin, token, function(error, resp) {
+                  if (error) {
+                    res.status(500).send({error: error});
+                  } else {
+                    res.status(200).json({success: 'Email sent'});
+                  }
+                });
+              }
+            });
+          } else {
+            res.status(500).send({error: 'Account not found'});
+          }
+        });
+      } else {
+        res.status(500).send({error: 'Credential not found'});
+      }
+    });
+  });
+
+  app.use('/update-password', function(req, res, next) {
+    var user = req.body;
+    var url = config.apiServer + config.apiEndpoint;
+
+    pool.query('SELECT * FROM challenge WHERE token = ? AND created > DATE_SUB(NOW(), INTERVAL 24 HOUR)', [user.token], function(error, results, fields) {
+      if (error) res.status(500).send({error: 'There was an error finding the record'});
```

Comment:
This should say something like the token is invalid or expired, please try requesting a new reset.